### ISSUE TITLE
Debian package release v1.8.7 fails to start with systemd

### ISSUE THREAD

**Issue by rda0** [influxdata/influxdb#21933]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. `wget https://dl.influxdata.com/influxdb/releases/influxdb_1.8.7_amd64.deb`
2. `dpkg -i influxdb_1.8.7_amd64.deb`
3. `systemctl restart influxdb.service`

__Expected behavior:__
Systemd is able to restart the service.

__Actual behavior:__
Systemd fails to execute the script as user `influxdb`:

```
ls -la /usr/lib/influxdb/scripts/influxd-systemd-start.sh
-rwxr--r-- 1 root root  520 Jul 21 21:55 influxd-systemd-start.sh*
```

__Environment info:__

* System info: Tested on Debian 9 stretch and 10 buster
* InfluxDB version: v1.8.7

__Logs:__

```
Jul 23 11:57:56 phd-test-influx systemd[1]: Starting InfluxDB is an open-source, distributed, time series database...
Jul 23 11:57:56 phd-test-influx systemd[2853]: influxdb.service: Failed to execute command: Permission denied
Jul 23 11:57:56 phd-test-influx systemd[2853]: influxdb.service: Failed at step EXEC spawning /usr/lib/influxdb/scripts/influxd-systemd-start.sh: Permission denied
Jul 23 11:57:56 phd-test-influx systemd[1]: influxdb.service: Control process exited, code=exited, status=203/EXEC
Jul 23 11:57:56 phd-test-influx systemd[1]: influxdb.service: Failed with result 'exit-code'.
Jul 23 11:57:56 phd-test-influx systemd[1]: Failed to start InfluxDB is an open-source, distributed, time series database.
Jul 23 11:57:56 phd-test-influx systemd[1]: influxdb.service: Service RestartSec=100ms expired, scheduling restart.
Jul 23 11:57:56 phd-test-influx systemd[1]: influxdb.service: Scheduled restart job, restart counter is at 1.
Jul 23 11:57:56 phd-test-influx systemd[1]: Stopped InfluxDB is an open-source, distributed, time series database.
```

__Steps to fix:__

```sh
chmod +x /usr/lib/influxdb/scripts/influxd-systemd-start.sh
systemctl restart influxdb.service
```

**Comment by tkaehn** [comment]
Same problem here. Would be great to see this fixed.

**Comment by ieguiguren** [comment]
Same issue on 
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal

With package:
 Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  influxdb       1.8.7-1      amd64        Distributed time-series database.

@rda0 fix works like a charm (thanx, Sven)

**Comment by Nanouk171** [comment]
System info: Tested on Debian 10 Buster 10.10
InfluxDB version: v1.8.7

I change the permission on /usr/lib/influxdb/scripts/influxd-systemd-start.sh with chmod +x.
The process start with systemd but not correctly:
# systemctl start influxdb.service 

Job for influxdb.service failed because a timeout was exceeded.
See "systemctl status influxdb.service" and "journalctl -xe" for details.
zsh: exit 1     systemctl start influxdb.service

Influxdb server is up: 

influxdb 26201  0.0  0.0   6728  3204 ?        Ss   11:38   0:00 /bin/bash -e /usr/lib/influxdb/scripts/influxd-systemd-start.sh
influxdb 26202 44.6  1.7 2802616 285056 ?      Sl   11:38   0:05  \_ /usr/bin/influxd -config /etc/influxdb/influxdb.conf
influxdb 26278  0.0  0.0   5260   744 ?        S    11:38   0:00  \_ sleep 1

But with telegraf client, I have many errors like that:

Jul 29 16:02:31 polux telegraf[935]: 2021-07-29T14:02:31Z E! [outputs.influxdb] When writing to [https://xxx.xxx.xxx.xxx:8086]: failed doing req: Post "https://xxx.xxx.xxx.xxx:8086/write?db=telegraf": dial tcp xxx.xxx.xxx.xxx:8086: connect: connection refused
Jul 29 16:04:01 polux telegraf[935]: 2021-07-29T14:04:01Z E! [outputs.influxdb] When writing to [https://xxx.xxx.xxx.xxx:8086]: failed doing req: Post "https://xxx.xxx.xxx.xxx:8086/write?db=telegraf": dial tcp xxx.xxx.xxx.xxx:8086: connect: connection refused

**Comment by gribnut** [comment]
Issue exists in both deb and rpm packages

**Comment by danpawlik** [comment]
Workarround: https://github.com/influxdata/influxdb/issues/21967#issuecomment-889695265

**Comment by basking-in-the-sun2000** [comment]
+1

**Comment by RolandRosenfeld** [comment]
Wouldn't it be a good idea to release a fix in a new version as soon as possible?
It's really a very bad user experience that upgrading from 1.8.6 to 1.8.7 breaks a running installation (at least every Debian installation, and if I got the comment about RPM above right, also all RPM installations).
Not every user will find the workaround in minutes.
It's unbelievable to me, that this issue is open for 9 days now without a fix being published nor 1.8.6-1 available in the repository for a downgrade...

**Comment by AndreasUfert** [comment]
Generally, it would be a good idea if older versions were still available in the repos. It would also help in staging scenarios where dev environments get an update while production environments still run a version that's known to be running perfectly.

**Comment by codyshepherd** [comment]
Hi there. [v1.8.9 was released today](https://github.com/influxdata/influxdb/releases/tag/v1.8.9) with a fix for this issue.

**Comment by codyshepherd** [comment]
This issue is duplicated by #22004

**Comment by Marci24h** [comment]
same issue here after upgrade from 1.8.6-1 to 1.8.7-1 on debian 10 (buster)
i need to make the start script executable.
`chmod +x /usr/lib/influxdb/scripts/influxd-systemd-start.sh`
i wrote about the issue on my small blog: [](https://greier.eu/2021/08/influxdb-startet-nicht-mehr/)

after upgrading to 1.8.9-1 the service stuck in a restart loop. but the executable issue was fixed ;)
also cause by the `influxd-systemd-start.sh` script.
in line 25 a counter of 10 max attempts is spesified.
since the loop sleeps for 1 second the timeout for starting the service is about 10 seconds.

since my installation is a bit bigger, it takes a little bit longer (14 seconds despite ssds). so i need to increase the value.

this problem is a bit similar to issue #21967 and #22110

**Comment by fonske** [comment]
@Marci24h 
This was the solution for me.
changed the max_attempts=10 to max_attempts=20 in
/usr/lib/influxdb/scripts/influxd-systemd-start.sh

and then 
systemctl restart influxdb.service

Checked by
systemctl status influxdb.service
"running" instead of "starting"

thank you.

**Comment by AndreasUfert** [comment]
I don't like the idea to patch a file that comes with a package. I don't like the idea of polling a certain amount of time either where nobody can tell how long it takes in a particular environment.

If we really have to live with that polling loop thing, at least make it a parameter in /etc/defaults/influxdb.

**Comment by rotanid** [comment]
> An rc (release candidate) has been built for 1.8.10 today with further fixes for this issue, e.g.: https://s3.amazonaws.com/dl.influxdata.com/influxdb/releases/influxdb_1.8.10rc0_amd64.deb, https://s3.amazonaws.com/dl.influxdata.com/influxdb/releases/influxdb-1.8.10rc0.x86_64.rpm

working for me, thanks! probably #22076 was the last piece.
there are many related issues which can be closed as soon as 1.8.10 is released :)

**Comment by codyshepherd** [comment]
1.8.10 includes a fix for this issue, and [has been released](https://github.com/influxdata/influxdb/releases/tag/v1.8.10).

**Comment by Marci24h** [comment]
For me the upgrade to version 1.8.10 worked without any problems.
systemd service is active and running ;)

**Comment by codyshepherd** [comment]
I'll close this bug as fixed in 1.8.10. Please open a new bug for any new issue that arises in 1.8.10.

**Comment by Ingo-Knito** [comment]
I leave here a new comment, as i have exact the problem here in 1.8.10 - and don't knw how to solve this issue.

On a raspi 3B+ i have installed Node-Red, mySQL, Grafana, ... The system was completely new installed after a non-working start of influxdb. Now it ran about a week and i wanted to activate Flux. After the stop service, influx does not start anymore.

Yes i've waited over an hor. No change.

The 'chmod +x /usr/lib/influxdb/scripts/influxd-systemd-start.sh' did not help.

`Job for influxdb.service failed because a timeout was exceeded.
See "systemctl status influxdb.service" and "journalctl -xe" for details.`

`pi@raspberrypi:~ $ systemctl status influxdb.service
● influxdb.service - InfluxDB is an open-source, distributed, time series database
     Loaded: loaded (/lib/systemd/system/influxdb.service; enabled; vendor preset: enabled)
     Active: activating (start) since Wed 2022-11-02 10:15:39 CET; 23s ago
       Docs: https://docs.influxdata.com/influxdb/
Cntrl PID: 9144 (influxd-systemd)
      Tasks: 14 (limit: 1596)
        CPU: 3.218s
     CGroup: /system.slice/influxdb.service
             ├─9144 /bin/bash -e /usr/lib/influxdb/scripts/influxd-systemd-start.sh
             ├─9145 /usr/bin/influxd -config /etc/influxdb/influxdb.conf
             └─9242 sleep 1

Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.509153Z lvl=info msg="Opened file" log_id=0duuqLV0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/3
Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.683797Z lvl=info msg="Opened shard" log_id=0duuqLV0000 service=store trace_id=0duuqLuW000 op_name=tsdb_open index_version=inmem
Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.684683Z lvl=info msg="Opened shard" log_id=0duuqLV0000 service=store trace_id=0duuqLuW000 op_name=tsdb_open index_version=inmem
Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.686606Z lvl=info msg="Opened file" log_id=0duuqLV0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/3
Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.687099Z lvl=info msg="Opened file" log_id=0duuqLV0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/3
Nov 02 10:15:59 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.920619Z lvl=info msg="Opened shard" log_id=0duuqLV0000 service=store trace_id=0duuqLuW000 op_name=tsdb_open index_version=inmem
Nov 02 10:16:00 raspberrypi influxd-systemd-start.sh[9145]: ts=2022-11-02T09:15:59.923709Z lvl=info msg="Opened file" log_id=0duuqLV0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/3
Nov 02 10:16:00 raspberrypi influxd-systemd-start.sh[9144]: InfluxDB API unavailable after 20 attempts...
Nov 02 10:16:01 raspberrypi influxd-systemd-start.sh[9144]: InfluxDB API unavailable after 21 attempts...
Nov 02 10:16:02 raspberrypi influxd-systemd-start.sh[9144]: InfluxDB API unavailable after 22 attempts...
`

`pi@raspberrypi:~ $ journalctl -xe
Nov 02 10:19:19 raspberrypi Node-RED[370]: }
Nov 02 10:19:19 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 26 attempts...
Nov 02 10:19:19 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:19.638994Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:19 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:19.640623Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:19 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:19.928833Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:19 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:19.930589Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.108211Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.108355Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.109920Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.109929Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 27 attempts...
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.818435Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:20 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:20.825336Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:21 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 2). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:21 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:21 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:21 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:21 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:21 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:21 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:21 raspberrypi Node-RED[370]: }
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.416343Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.419908Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 28 attempts...
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.594215Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.594124Z lvl=info msg="Opened shard" log_id=0duv16Q0000 service=store trace_id=0duv16qW000 op_name=tsdb_open index_version=inme>
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.596081Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:21 raspberrypi influxd-systemd-start.sh[9791]: ts=2022-11-02T09:19:21.598045Z lvl=info msg="Opened file" log_id=0duv16Q0000 engine=tsm1 service=filestore path=/var/lib/influxdb/data/DBGas/autogen/>
Nov 02 10:19:22 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 5). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:22 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:22 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:22 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:22 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:22 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:22 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:22 raspberrypi Node-RED[370]: }
Nov 02 10:19:22 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 29 attempts...
Nov 02 10:19:23 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 4). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:23 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:23 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:23 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:23 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:23 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:23 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:23 raspberrypi Node-RED[370]: }
Nov 02 10:19:23 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 30 attempts...
Nov 02 10:19:24 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 4). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:24 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:24 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:24 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:24 raspberrypi Node-RED[370]: }
Nov 02 10:19:24 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 4). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:24 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:24 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:24 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:24 raspberrypi Node-RED[370]: }
Nov 02 10:19:24 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 31 attempts...
Nov 02 10:19:24 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 3). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:24 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:24 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:24 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:24 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:24 raspberrypi Node-RED[370]: }
Nov 02 10:19:25 raspberrypi influxd-systemd-start.sh[9790]: InfluxDB API unavailable after 32 attempts...
Nov 02 10:19:26 raspberrypi Node-RED[370]: WARN: Write to InfluxDB failed (attempt: 2). Error: connect ECONNREFUSED 127.0.0.1:8086
Nov 02 10:19:26 raspberrypi Node-RED[370]:     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
Nov 02 10:19:26 raspberrypi Node-RED[370]:   errno: -111,
Nov 02 10:19:26 raspberrypi Node-RED[370]:   code: 'ECONNREFUSED',
Nov 02 10:19:26 raspberrypi Node-RED[370]:   syscall: 'connect',
Nov 02 10:19:26 raspberrypi Node-RED[370]:   address: '127.0.0.1',
Nov 02 10:19:26 raspberrypi Node-RED[370]:   port: 8086
Nov 02 10:19:26 raspberrypi Node-RED[370]: }
`
